core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -305,7 +325,7 @@ class ReplicaManager(val config: KafkaConfig,
                     responseCallback: Map[TopicPartition, PartitionResponse] => Unit) {
 
     if (isValidRequiredAcks(requiredAcks)) {
-      val sTime = time.milliseconds
+      val sTime = if (logger.isDebugEnabled) time.milliseconds else 0
```

Comment:
It will call `time.milliseconds` because it is an argument passed to the `debug()`.